### PR TITLE
Add USB to Serial/JTAG support for REPL

### LINKED ISSUE #6030: Connect serial on the ESP32-C3 to the native USB converter
That way it's available on the native USB plug.

### PR DESCRIPTION
**Pull request by tannewt**
Adds Adafruit QT Py C3 board that uses it. Also revamps size
check script to work for S3 and C3 as well.

Fixes #6030

### CODE REVIEW

**Comment by tannewt** [comment]
> Can you move supervisor/serial.c out of the port? If needed, have e.g., `port_bytes_available()` and such functions that are port-specific.

Done in the latest commit. :crossed_fingers: all of the boards build.

**Comment by tannewt** [comment]
@jepler I spot checked failures and they all build for me. I can't load the files to know why they failed here. I suspect it's not my problem because they all fail in 4 minutes or less.

**Comment by tannewt** [comment]
@jepler CI passed!

File: supervisor/shared/serial.c [hunk 1]
```diff
@@ -88,6 +88,26 @@ int debug_uart_printf(const char *fmt, ...) {
     #endif
 }
 
+MP_WEAK void port_serial_init(void) {
```
Comment (jepler) [inline supervisor/shared/serial.c:91]: nice use of MP_WEAK

**Review by jepler — APPROVED** [review]
Thank you! didn't test though

**Pull request merged**